templates/extensions/subgraph/packages/subgraph/scripts/abi_copy.ts

```diff
@@ -1,112 +1,120 @@
   networkName: string
-) {
-  try {
-    const graphConfigPath = `${GRAPH_DIR}/networks.json`;
-    let graphConfig = "{}";
-    try {
-      if (fs.existsSync(graphConfigPath)) {
-        graphConfig = fs.readFileSync(graphConfigPath).toString();
-      }
-    } catch (e) {
-      console.log(e);
-    }
+) => {
+  const graphConfigPath = `${GRAPH_DIR}/networks.json`;
+  const abisDir = `${GRAPH_DIR}/abis`;
+  const configContent = fs.existsSync(graphConfigPath)
+    ? fs.readFileSync(graphConfigPath, "utf8")
+    : "{}";
+  const graphConfig = JSON.parse(configContent);
 
-    let graphConfigObject = JSON.parse(graphConfig);
-    if (!(networkName in graphConfigObject)) {
-      graphConfigObject[networkName] = {};
-    }
-    if (!(contractName in graphConfigObject[networkName])) {
-      graphConfigObject[networkName][contractName] = {};
-    }
-    graphConfigObject[networkName][contractName].address =
-      contractObject.address;
+  graphConfig[networkName] = graphConfig[networkName] || {};
+  graphConfig[networkName][contractName] = { address: contractObject.address };
 
-    fs.writeFileSync(
-      graphConfigPath,
-      JSON.stringify(graphConfigObject, null, 2)
-    );
-    if (!fs.existsSync(`${GRAPH_DIR}/abis`)) fs.mkdirSync(`${GRAPH_DIR}/abis`);
-    fs.writeFileSync(
-      `${GRAPH_DIR}/abis/${networkName}_${contractName}.json`,
-      JSON.stringify(contractObject.abi, null, 2)
-    );
+  fs.writeFileSync(graphConfigPath, JSON.stringify(graphConfig, null, 2));
 
-    return true;
-  } catch (e) {
-    console.log(
-      "Failed to publish " + chalk.red(contractName) + " to the subgraph."
-    );
-    console.log(e);
-    return false;
-  }
-}
+  if (!fs.existsSync(abisDir)) fs.mkdirSync(abisDir);
+  fs.writeFileSync(
+    `${abisDir}/${networkName}_${contractName}.json`,
+    JSON.stringify(contractObject.abi, null, 2)
+  );
+};
 
-const DEPLOYED_CONTRACTS_FILE = "../nextjs/contracts/deployedContracts.ts";
 async function main() {
-  const fileContent = fs.readFileSync(DEPLOYED_CONTRACTS_FILE, "utf8");
-
-  const pattern = /const deployedContracts = ({[^;]+}) as const;/s;
-  const match = fileContent.match(pattern);
+  try {
+    const deployedContractsContent = readConfigFile(DEPLOYED_CONTRACTS_FILE);
+    const externalContractsContent = readConfigFile(EXTERNAL_CONTRACTS_FILE);
 
-  if (!match || !match[1]) {
-    throw new Error(
-      `Failed to find deployedContracts in the ${DEPLOYED_CONTRACTS_FILE}`
+    const deployedContractsMatch = deployedContractsContent.match(
+      /const deployedContracts = ({[^;]+}) as const;/s
     );
-  }
-  const jsonString = match[1];
 
-  // Parse the JSON string
-  const deployedContracts = parseAndCorrectJSON(jsonString);
-  const localContracts = deployedContracts[31337];
+    // doing global match since we have example comment in extrenalContracts.ts
+    const externalContractsRegex =
+      /^const\s+externalContracts\s*=\s*({[\s\S]*?})\s*as\s*const;/gm;
```

Comment:
Not proud of this regex matching solution, but didn't find any simple / straight forwad to read externalContracts.ts & deployedContracts.ts  variables

But tested it nicely and seems to work for in different cases 

---

Alternate solutions : 

We can't import files directly from `packages/nextjs/contracts` since `tsconfig` in `pacakges/subgraph` doesn't know about it. 

When I tried adding "@se-2/nextjs" as dependency to `subgraph` package and directly importing `deployedContract.ts` I get this : 

<details>

<summary>
Error : 
</summary>

![Screenshot 2024-05-15 at 12 46 26 PM](https://github.com/scaffold-eth/create-eth/assets/80153681/d9b93ab7-4d72-4146-9d32-3311c3e2a84e)


So it's not able to resolve alias
</details>

Hacky solution, to solve above error https://github.com/technophile-04/testing-subgraph/commit/054926e4b8d45af1a9eb88158441497540fc1f69

But I think there are some problems, while adding `type: "module"` in nextjs package.json (because of which my lint fails) here is [related discussion ](https://github.com/vercel/next.js/issues/31518) 

We can explorer above solution more in maybe other PR, lol didn't wanted to have any side-effects to base nextjs stuff so kepts it as it and used regex to handle it

Other solution might be using some parser like bable which parse the AST and get the variables from the file